source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -194,15 +208,37 @@ A driver-defined wrapper around a single TCP/IP connection to an Endpoint. A Con
        *  when this connection was created
        */
       generation: number;
+
+      /**
+       * The current state of the Connection.
+       *
+       * Possible values are the following:
+       *   - "unestablished": The Connection is empty. It has been created but has not opened a socket or
+       *                      performed any I/O. Contributes to totalConnectionCount.
+       *
+       *   - "available":     The Connection has been established and is waiting in the pool to be checked
+       *                      out. Contributes to both totalConnectionCount and availableConnectionCount.
+       *
+       *   - "in use":        The Connection has been established, checked out from the pool, and has yet
+       *                      to be checked back in. Contributes to totalConnectionCount.
+       *
+       *   - "closed":        The Connection has had its socket closed and cannot be used for any future
+       *                      operations. Does not contribute to any connection counts.
+       *
+       * Note: this field is mainly used for the purposes of describing state
```

Comment:
I hadn't seen this done before in other specs, but having this "spec-only" enumeration of states helps me visualize the lifecycle of a connection, and it also helps make the spec language / psuedocode more precise.

Reply:
:+1: